### PR TITLE
V0.3 min

### PR DESCRIPTION
**Pull request by mpodwysocki**
Things to consider before you submit the PR:

* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Minimal update to use the `2020-06` API version, includes the `userId` and throttling changes.

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: NotificationHubs/src/com/windowsazure/messaging/Installation.java [hunk 1]
```diff
@@ -1,133 +1,366 @@
...
+
+    /**
+     * Adds a secondary tile to the installation template.
+     *
+     * @param tileName The name for the tile.
+     * @param tile     THe WNS secondary tile.
+     */
+    public void addSecondaryTile(String tileName, WnsSecondaryTile tile) {
+        if (this.templates == null) {
+            this.secondaryTiles = new HashMap<>();
```
Comment (marstr) [inline NotificationHubs/src/com/windowsazure/messaging/Installation.java:312]: What's the minimum version of Java that we allow? I know implicit type parameters aren't supported in older versions, but I can't remember the exact cutoff.

File: NotificationHubs/src/com/windowsazure/messaging/Notification.java [hunk 2]
```diff
@@ -6,245 +10,381 @@
...
+        Notification n = new Notification();
+        n.body = body;
+
+        n.headers.put("ServiceBusNotification-Format", "windows");
+
+        if (body.matches("^<toast[\\s\\S]*>[\\s\\S]+</toast>$"))
+            n.headers.put("X-WNS-Type", "wns/toast");
+        if (body.matches("^<tile[\\s\\S]*>[\\s\\S]+</tile>$"))
+            n.headers.put("X-WNS-Type", "wns/tile");
+        if (body.matches("^<badge[\\s\\S]*>[\\s\\S]+</badge>$"))
```
Comment (marstr) [inline NotificationHubs/src/com/windowsazure/messaging/Notification.java:40]: So we previously only looked for the opening tag, but would miss if it included any attributes. This now captures the entire body. Do we have corresponding tests for this regexp? Seems easy to get wrong.

Comment (marstr) [reply NotificationHubs/src/com/windowsazure/messaging/Notification.java:40]: For instance, in the case there were multiple "tile" instances in the body, the greedy match in the beginning would swallow all instances and turn it into one match.

File: NotificationHubs/src/com/windowsazure/messaging/Notification.java [hunk 3]
```diff
@@ -6,245 +10,381 @@
...
+        }
+
+        return n;
+    }
+
+    /**
+     * Utility method to set up a native notification for WNS. Sets the X-WNS-Type
+     * header to "wns/raw" in order of sending of raw notification.
+     *
+     * @param body The b ody for the Windows Raw Notification
```
Comment (marstr) [inline NotificationHubs/src/com/windowsazure/messaging/Notification.java:54]: nit, extra space here

File: NotificationHubs/src/com/windowsazure/messaging/Installation.java (hunk 1 above)
Comment (mpodwysocki) [reply NotificationHubs/src/com/windowsazure/messaging/Installation.java:312]: I've updated this to be Java 8+

File: NotificationHubs/src/com/windowsazure/messaging/Notification.java (hunk 2 above)
Comment (mpodwysocki) [reply NotificationHubs/src/com/windowsazure/messaging/Notification.java:40]: This didn't change from the previous release, only the spacing changed

File: NotificationHubs/src/com/windowsazure/messaging/Notification.java (hunk 3 above)
Comment (mpodwysocki) [reply NotificationHubs/src/com/windowsazure/messaging/Notification.java:54]: fixed

**Review by marstr — APPROVED** [review]

**Pull request merged**